### PR TITLE
Added Factory/Pool Documentation

### PR DESCRIPTION
**Pull request by davejrodriguez**
An initial attempt at documentation.

This PR will help others use Factories and Pools more effectively. Documentation in these markdown files should also be easily adapted to wiki pages if the maintainers choose to do so.

Though I have an open PR #159 for changes to these libraries, the documentation in _this_ PR should be valid for the libraries as they currently stand.

### CODE REVIEW

File: UOP1_Project/Assets/Scripts/Factory/DOCUMENTATION.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
+# Factory
+
+![Factory Diagram](https://i.imgur.com/8cleBXk.png)
```
Comment (spaceemotion) [inline UOP1_Project/Assets/Scripts/Factory/DOCUMENTATION.md:3]: Is there a reason for hosting this on imgur other than convenience? It seems like they're usually stored alongside the documentation contents (e.g. https://github.com/UnityTechnologies/ScriptableRenderPipeline/tree/master/com.unity.render-pipelines.lightweight/Documentation~/images)

Comment (davejrodriguez) [reply UOP1_Project/Assets/Scripts/Factory/DOCUMENTATION.md:3]: Just following the precedent set on this repo. The [main README](https://raw.githubusercontent.com/UnityTechnologies/open-project-1/main/README.md) images are all hosted on imgur. But I can re-export diagrams and put them wherever.

Comment (spaceemotion) [reply UOP1_Project/Assets/Scripts/Factory/DOCUMENTATION.md:3]: Ah, didn't look at existing images in this repo!

**Comment by ciro-unity** [comment]
Hey Dave, this is fantastic!

However, we want to use the Wiki here on Github as a source for documentation. I have created a few pages, and transferred the contents of this PR into that.
https://github.com/UnityTechnologies/open-project-1/wiki
What do you think?

**Comment by davejrodriguez** [comment]
Yes, my goal was to make something that could easily transfer to the wiki. Looks great!

Minor thing - I wouldn't consider Factory a sub-category of Pool. Someone could conceivably implement an IPool that does not use a factory. Or use factories by themselves for another system. But I suppose in the project's current state with Pool being the only library depending on Factory, it makes sense conceptually for now.

**Comment by ciro-unity** [comment]
Yeah, fair enough. I've removed the indentation. I'll frequently reorganise these topics, for the sake of clarity!

**Comment by davejrodriguez** [comment]
Closing as docs were integrated into the wiki.

**Pull request closed without merging**